packages/go_router/test/src/custom_transition_page_test.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+
+import '../go_router_test.dart';
+
+void main() {
+  testWidgets('CustomTransitionPage builds its child using transitionsBuilder',
+      (tester) async {
+    const child = HomeScreen();
+    final transition = CustomTransitionPage<void>(
+      transitionsBuilder: expectAsync4((_, __, ___, child) => child),
+      child: child,
+    );
+    final router = GoRouter(
+      routes: [
+        GoRoute(
+          path: '/',
+          pageBuilder: (_, __) => transition,
+        ),
+      ],
+    );
+    await tester.pumpWidget(
+      MaterialApp.router(
+        routeInformationParser: router.routeInformationParser,
+        routerDelegate: router.routerDelegate,
+        title: 'GoRouter Example',
+      ),
+    );
+    expect(find.byWidget(child), findsOneWidget);
+  });
+
+  test('NoTransitionPage does not apply any transition', () {
+    const homeScreen = HomeScreen();
+    const page = NoTransitionPage<void>(child: homeScreen);
+    const primaryAnimation = AlwaysStoppedAnimation<double>(0);
+    const secondaryAnimation = AlwaysStoppedAnimation<double>(1);
+    final widget = page.transitionsBuilder(
```

Comment:
This is testing implementation detail, can you write a widget test to make sure the there is no transition when push/pop the NoTransitionPage from the go_router?

Reply:
Sure, but I have no idea about how to do that and I would like some suggestions. I've tried using `tester.hasRunningAnimations` but it seems to return true for both cases (using CustomTransitionPage or NoTransitionPage).

Reply:
you can create a material app Something like this
```
MaterialApp(
   home: Navigator(
      pages: [NoTransitionPage<void>(...)],
    )
)
```
and then pump the same widget with an additional NoTransitionPage which will cause a push. Then you assert the location of the widget in the new page does not change within the duration of the transition.

after that, you can call pop on the navigator, and again expect the widget is removed from the screen right away.